pkg/storage/wal/manager_test.go

```diff
@@ -90,25 +86,103 @@ func TestManager_Append(t *testing.T) {
 	require.EqualError(t, res.Err(), "failed to flush")
 }
 
-// This test asserts that Append operations return ErrFull if all segments
-// are full and waiting to be flushed.
-func TestManager_Append_ErrFull(t *testing.T) {
+func TestManager_AppendMaxAge(t *testing.T) {
 	m, err := NewManager(Config{
+		MaxAge:         100 * time.Millisecond,
+		MaxSegments:    1,
+		MaxSegmentSize: 8 * 1024 * 1024, // 8MB
+	}, NewMetrics(nil))
+	require.NoError(t, err)
+
+	// Append 1B of data.
+	lbs := labels.Labels{{Name: "a", Value: "b"}}
+	entries := []*logproto.Entry{{Timestamp: time.Now(), Line: "c"}}
+	res, err := m.Append(AppendRequest{
+		TenantID:  "1",
+		Labels:    lbs,
+		LabelsStr: lbs.String(),
+		Entries:   entries,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, res)
+
+	// The segment that was just appended to has neither reached the maximum
+	// age nor maximum size to be flushed.
+	require.Equal(t, 1, m.available.Len())
+	require.Equal(t, 0, m.pending.Len())
+
+	// Wait 100ms and append some more data.
+	time.Sleep(100 * time.Millisecond)
```

Comment:
You could probably set the Timestamp values to avoid the sleep?

Reply:
Looking to find an alternative to https://github.com/benbjohnson/clock which is archived. I might end up using https://github.com/jonboulle/clockwork.

Reply:
I will open another PR for this.